controllers/progresses.js

```diff
@@ -107,8 +108,38 @@ const createProgress = async (req, res) => {
  */
 
 const getProgress = async (req, res) => {
+  const { dev, page = 0, size = 100, type, userId, taskId } = req.query;
   try {
     const data = await getProgressDocument(req.query);
+    if (dev) {
+      const paginatedProgressDocs = await getPaginatedProgressDocument(req.query);
+      const limit = parseInt(size, 10);
+      const offset = parseInt(page, 10) * limit;
```

Comment:
Are we not validating the type of the data at validation layer?

Reply:
I added this just for extra safety to ensure the values are parsed correctly, even though the validator checks the data type